app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt

```diff
@@ -91,9 +93,28 @@ class NewMessageViewModel : ViewModel() {
         }
     }
 
-    private fun MutableRealm.createDraft(): String {
+    private fun MutableRealm.createDraft(draftMode: DraftMode, previousMessageUid: String?): String {
         return Draft()
             .initLocalValues(priority = Priority.NORMAL)
+            .apply {
+                previousMessageUid?.let { MessageController.getMessage(it, this@createDraft) }?.let { previousMessage ->
+
+                    previousMessage.msgId.let {
+                        this.inReplyTo = it
+                        this.references = it
+                    }
+
+                    when (draftMode) {
+                        DraftMode.REPLY, DraftMode.REPLY_ALL -> this.inReplyToUid = previousMessageUid
+                        DraftMode.FORWARD -> this.forwardedUid = previousMessageUid
+                        DraftMode.NEW_MAIL -> Unit
+                    }
+
+                    this.to = previousMessage.from
+                    if (draftMode == DraftMode.REPLY_ALL) this.cc = previousMessage.to.union(previousMessage.cc).toRealmList()
+                    previousMessage.subject?.let { this.subject = "Re: $it" }
+                }
+            }
             .also { DraftController.upsertDraft(it, this) }
             .localUuid
```

Comment:
You should refactor this whole part. Too many apply, let and also at the same time. It's very difficult to read.

Reply:
I jsut see it : https://github.com/Infomaniak/android-mail/pull/321/files